machine/parser_test.go

```diff
@@ -109,3 +108,10 @@ func Test_Machine_Parser_TaskFn(t *testing.T) {
 	assert.Equal(t, err, nil)
 	assert.NoError(t, sm.Validate())
 }
+
+func Test_Machine_Parser_Map(t *testing.T) {
+	sm, err := ParseFile("../examples/map.json")
```

Comment:
We should test the output here to make sure it is correct. 
This will involve making sure the Task state inside the iterator is correctly being called.